nucypher/characters/control/specifications/exceptions.py

```diff
@@ -0,0 +1,14 @@
+class SpecificationError(ValueError):
+    """The protocol request is completely unusable"""
+
+class MissingField(SpecificationError):
+    """The protocol request cannot be deserialized because it is missing required fields"""
+
+class InvalidInputField(SpecificationError):
+    """Response data does not match the input specification"""
+
+class InvalidOutputField(SpecificationError):
+    """Response data does not match the output specification"""
+
+class InvalidArgumentCombo(SpecificationError):
```

Comment:
This is interesting.  We've often discussed how to formalize "xor" arguments - we have different mutually exclusivity logic in different parts of the codebase.  Perhaps this is the beginning of something normal.